### PR TITLE
feat(components/atom/polymorphicElement): implement polymorphic element

### PR DESCRIPTION
**Pull request by tonyghiani**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [ ] Refactor

### Description, Motivation and Context
In order to improve the DX when using the library, we want to expose a polymorphic element to render a specific HTML tag when required.

### CODE REVIEW

File: components/atom/polymorphicElement/demo/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import PolymorphicElement from 'components/atom/polymorphicElement/src'
+
+function StrongText(props) {
+  return <span style={{fontWeight: 700}} {...props} />
+}
+
+function Demo() {
+  return (
+    <>
```
Comment (andresin87) [inline components/atom/polymorphicElement/demo/index.js]:
```js
<div className="sui-StudioPreview">
  <H1>PolymorphicElement</H1>
  <Paragraph>
    PolymorphicElement is ...
  </Paragraph>
  <Article className="DemoAtomPolymorphicElement">
    <H2>Default</H2>
    <Paragraph>By default, the component assigns an span as a default tag element</Paragraph>
    <PolymorphicElement>
      Polymorphic item rendering a span by default.
    </PolymorphicElement>
    <Paragraph>
      User can change its tag using the <Code>as</Code> (elementType) prop</Paragraph>
    <PolymorphicElement as="h1">
      Polymorphic item rendering as a h1 tag.
    </PolymorphicElement>
    <PolymorphicElement as="div">
      Polymorphic item rendering as a div tag.
    </PolymorphicElement>
    <PolymorphicElement as={StrongText}>
      Polymorphic item rendering as an existing component.
    </PolymorphicElement>
  </Article
</div>
```

File: components/atom/polymorphicElement/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@s-ui/react-atom-polymorphic-element",
+  "version": "1.0.0",
+  "description": "",
+  "main": "lib/index.js",
+  "scripts": {
+    "prepare": "npm run build:js && npm run build:styles",
```
Comment (andresin87) [inline components/atom/polymorphicElement/package.json:7]: "prepare": "rimraf ./lib && npm run build:js && npm run build:styles",

Comment (andresin87) [inline components/atom/polymorphicElement/package.json]: "prepare": "rimraf ./lib && npm run build:js && npm run build:styles",

**Review by manjarcode — APPROVED** [review]

File: components/atom/polymorphicElement/demo/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+function StrongText(props) {
+  return <span style={{fontWeight: 700}} {...props} />
+}
+
+function Demo() {
+  return (
+    <div className="sui-StudioPreview">
+      <H1>PolymorphicElement</H1>
+      <Paragraph>PolymorphicElement is ...</Paragraph>
```
Comment (andresin87) [inline components/atom/polymorphicElement/demo/index.js]: PolymorphicElement is what ... 😅

**Review by andresin87 — APPROVED** [review]

**Pull request merged**